### PR TITLE
Fix 801652: New Editor: broken history of references navigation

### PR DESCRIPTION
**Pull request by DavidKarlas**
* main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewContent.cs:
 * main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewContent.TextViewNavigationPoint.cs:
 * main/src/addins/MonoDevelop.TextEditor/MonoDevelop.TextEditor/TextViewExtensions.cs:
In past support for new editor was added inside new editor by implementing `INavigation` which is debatable better or not approach. But since TextFileNavigationPoint is there and can be shared between old and new editor, I decided to remove `TextViewContent.TextViewNavigationPoint.cs` file and use `TextFileNavigationPoint.cs`.
 * main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.FindInFiles/SearchResultWidget.cs:
This file mostly just follows API change and also waits for ITextView to be added to document before calling navigation
 * main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/TextFileNavigationPoint.cs:
Changes in this file are migration from old editor API to ITextView
 * main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/NavigationHistoryService.cs:
Biggest change here which fixes also old editor when it's opened for first time is usage of `document.RunWhenContentAdded<ITextView> (textView => {` because otherwise `.Editor` was null while content was still generating hence none of events were registered and navigation didn't work. With this change and migration to `ITextView` both new and old editor work. Rest of non-obvious changes are commented.

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/NavigationHistoryService.cs [hunk 1]
```diff
@@ -199,19 +201,19 @@ NavigationPoint GetNavPointForDoc (Document doc, bool forClosedHistory)
 					return point;
 			}
 
-			#pragma warning disable CS0618, CS0612 // Type or member is obsolete
-			var editBuf = doc.Editor;
-			if (editBuf != null) {
+			if (doc.GetContent<ITextView> () is ITextView textView) {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Navigation/NavigationHistoryService.cs]:
```suggestion
			if (doc.GetContent<ITextView> (forActiveView:true) is ITextView textView) {
```

**Comment by DavidKarlas** [comment]
@sevoku done

**Comment by abock** [comment]
This is somehow failing in the `Make dmg` step:

```
warning MM0009: Error while loading assemblies: git2-572e4d8: Error while loading assemblies: /Users/vsts/agent/2.152.1/work/1/s/md-addins/MacSetup/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/VersionControl/lib/win32/x64/git2-572e4d8.dll
	Error while loading assemblies: /Users/vsts/agent/2.152.1/work/1/s/md-addins/MacSetup/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/VersionControl/lib/win32/x64/git2-572e4d8.dll
	Format of the executable (.exe) or library (.dll) is invalid.
warning MM0009: Error while loading assemblies: git2-572e4d8: Error while loading assemblies: /Users/vsts/agent/2.152.1/work/1/s/md-addins/MacSetup/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/VersionControl/lib/win32/x64/git2-572e4d8.dll
	Error while loading assemblies: /Users/vsts/agent/2.152.1/work/1/s/md-addins/MacSetup/Visual Studio.app/Contents/Resources/lib/monodevelop/AddIns/VersionControl/lib/win32/x64/git2-572e4d8.dll
	Format of the executable (.exe) or library (.dll) is invalid.
```

**Comment by abock** [comment]
Rebased

**Comment by abock** [comment]
@slluis something very odd is going on here ... the DMG produced for this is 2.9GB ...

**Comment by abock** [comment]
For reference... relative to some other recent PR build downloads I had:

image

The dmg also does not extract at all. It's very corrupt.

**Comment by Therzok** [comment]
@rodrigomoya has been doing some CI work recently

**Comment by abock** [comment]
Maybe related to #7830 😄

**Comment by rodrmoya** [comment]
hmm, the breakage I added was not even allowing the DMG to be generated, now it's building again on non-mojave bots, so yes, the size of that DMG is completely wrong :( with #7830 though, same fix in https://github.com/xamarin/md-addins/pull/4702 the DMG produced is around 500/600 MB, back to normal, so that should fix it indeed, I hope

**Comment by directhex** [comment]
Old process: generate 3gb image, copy files in, convert to compressed image 

New process: create staging directory, create compressed image of its contents 

3G images are uncompressed images which somehow got uploaded anyway as part of the old process

**Comment by slluis** [comment]
@monojenkins backport master

**Comment by sevoku** [comment]
@monojenkins backport master
@monojenkins backport release-8.2

**Pull request merged**